File: Linear Control Course/Chapter5/Lesson3/Chapter5_Lesson3.cpp
#include <iostream>
#include <cmath>

int main() {
    const double K = 1.0;
    const double tau = 0.4;
    const double t_end = 5.0 * tau;
    const int N = 200;
    const double dt = t_end / static_cast<double>(N - 1);

    for (int k = 0; k != N; ++k) {
        double t = k * dt;
        double y = K * (1.0 - std::exp(-t / tau));
        std::cout << t << "," << y << std::endl;
    }
    return 0;
}

File: Linear_Control_Course/Chapter5/Lesson3/Chapter5_Lesson3.java
import java.util.Locale;

public class FirstOrderStep {
    public static void main(String[] args) {
        double K = 1.0;
        double tau = 0.4;
        double tEnd = 5.0 * tau;
        int N = 200;
        double dt = tEnd / (N - 1);

        Locale.setDefault(Locale.US);
        for (int k = 0; k != N; ++k) {
            double t = k * dt;
            double y = K * (1.0 - Math.exp(-t / tau));
            System.out.printf("%.4f,%.4f%n", t, y);
        }
    }
}
